docs/operator-manual/secret-management.md

```diff
@@ -1,29 +1,43 @@
 # Secret Management
 
-Argo CD is un-opinionated about how secrets are managed. There are many ways to do it, and there's no one-size-fits-all solution.
+There are two general ways to populate secrets when doing GitOps: on the destination cluster, or in Argo CD during 
+manifest hydration. We strongly recommend the former, as it is more secure and provides a better user experience.
 
-Many solutions use plugins to inject secrets into the application manifests. See [Mitigating Risks of Secret-Injection Plugins](#mitigating-risks-of-secret-injection-plugins)
-below to make sure you use those plugins securely.
+For discussion, see [#1364](https://github.com/argoproj/argo-cd/issues/1364).
 
-Here are some ways people are doing GitOps secrets:
+## Destination Cluster Secret Management
 
-* [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets)
-* [External Secrets Operator](https://github.com/external-secrets/external-secrets)
-* [Hashicorp Vault](https://www.vaultproject.io)
-* [Bank-Vaults](https://bank-vaults.dev/)
-* [Helm Secrets](https://github.com/jkroepke/helm-secrets)
-* [Kustomize secret generator plugins](https://github.com/kubernetes-sigs/kustomize/blob/fd7a353df6cece4629b8e8ad56b71e30636f38fc/examples/kvSourceGoPlugin.md#secret-values-from-anywhere)
-* [aws-secret-operator](https://github.com/mumoshu/aws-secret-operator)
-* [KSOPS](https://github.com/viaduct-ai/kustomize-sops#argo-cd-integration)
-* [argocd-vault-plugin](https://github.com/argoproj-labs/argocd-vault-plugin)
-* [argocd-vault-replacer](https://github.com/crumbhole/argocd-vault-replacer)
-* [Kubernetes Secrets Store CSI Driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver)
-* [Vals-Operator](https://github.com/digitalis-io/vals-operator)
-* [argocd-secret-replacer](https://github.com/mmalyska/argocd-secret-replacer)
+In this approach, secrets are populated on the destination cluster, and Argo CD does not need to directly manage them.
+[Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets) and [External Secrets Operator](https://github.com/external-secrets/external-secrets)
+are examples of this style of secret management.
```

Comment:
I'm sure we could list lost but I think just listing two examples is great. I might add the csi-driver because it's between these three you represent all the major styles of secrets management. https://github.com/kubernetes-sigs/secrets-store-csi-driver/tree/v1.4.0

Reply:
Added CSI! I thought about adding a link to Andrew's talk, but left it out since he explicitly endorses plugin-style secrets management. Agreed that we need an updated talk.